### PR TITLE
ref(analytics): Move integration analytics into submodules

### PR DESCRIPTION
**Pull request by leeandher**
This PR organizes the integration analytics into a directory. They were getting extremely long, and with the linting rule to sort dictionary keys, it was hard to parse which ones are related to one another.

This PR introduces some organization changes to the integration analytics. The current file is pretty long and the new linting rule which sorts keys makes it hard to group analytics that map to similar payloads or features. As a prototype, I've created submodules for `StacktraceLink` events and `Codeowners` events to show what I think is a better pattern.

The added benefit of this new pattern is that when using the analytics function, you don't have to rewrite (and possibly typo) the analytics key. With it inside an enum, it forces the user to use the exact correct key. However, to be backwards compatible (and avoid changing a bunch of files in this PR, I added a literal type so that using a string literal is still valid. I changed one example in `stacktraceLink.tsx` to show what I'd like our analytics to look like in the future.

No functional changes or new event types in this PR.

My goal is to use this new sub directory structure to create a new set of analytic events for the example app.

### CODE REVIEW

File: static/app/utils/analytics/integrations/index.ts [hunk 1]
```diff
@@ -125,29 +107,12 @@ export const integrationEventMap: Record<IntegrationAnalyticsKey, string> = {
...
   'integrations.serverless_functions_viewed': 'Integrations: Serverless Functions Viewed',
   'integrations.installation_input_value_changed':
     'Integrations: Installation Input Value Changed',
   'integrations.serverless_function_action': 'Integrations: Serverless Function Action',
   'integrations.cloudformation_link_clicked': 'Integrations: CloudFormation Link Clicked',
   'integrations.switch_manual_sdk_setup': 'Integrations: Switch Manual SDK Setup',
-  'integrations.code_owners_cta_setup_clicked':
-    'Integrations: Code Owners CTA Setup Clicked',
-  'integrations.code_owners_cta_docs_clicked':
-    'Integrations: Code Owners CTA Setup Clicked',
```
Comment (leeandher) [inline static/app/utils/analytics/integrations/index.ts:149]: To prove my point about how these can be missed, take a look at how these two events mistakenly have the same event name, likely because they were copy pasted while being created. It's really easy to miss this stuff!

Comment (leeandher) [reply static/app/utils/analytics/integrations/index.ts:149]: Enums might not have prevented this exact error from happening but, if someone were to copy `integrations.code_owners_cta_docs_clicked` from here because they read "CTA Setup Clicked", they could record the wrong analytics interaction in the product. Having to access CodeownersEvents.DOCS_CTA is a little bit more explicit, and if we want to change the analytics key, its a single line change.

**Review by ceorourke — APPROVED** [review]
Approving cause this looks fine, but is the ideal situation that each type of analytics lives in its own file, and you're not touching the other ones since they're not ones we own? This leaves us in a state where it's half using one system and half using another, and I doubt anyone (else) is going to go and update it to match the newer way. It's not a huge concern, just something I wanted to point out.

**Comment by leeandher** [comment]
@ceorourke Totally agree that it makes for some unique behaviour on integration analytics, but TBH I think that is already the case with most other analytics able to use `trackAdvancedAnalyticsEvent`, the generic wrapper, but integrations have to use `trackIntegrationAnalytics` since we need the organization every time.

I prefer this system of splitting analytics by file rather than having humongous files where most of the events look the same and making changes becomes risky since the analytics keys are rewritten in a bunch of places.
